Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,779 @@
+        SetIndex& triangleIndices = *d_triangleIndices.beginWriteOnly();
+        SetIndex& quadIndices = *d_quadIndices.beginWriteOnly();
+        SetIndex& tetrahedronIndices = *d_tetrahedronIndices.beginWriteOnly();
+        SetIndex& hexahedronIndices = *d_hexahedronIndices.beginWriteOnly();
+        SetIndex& indicesOut = *d_indicesOut.beginWriteOnly();
+        SetIndex& edgeOutIndices = *d_edgeOutIndices.beginWriteOnly();
+        SetIndex& triangleOutIndices = *d_triangleOutIndices.beginWriteOnly();
+        SetIndex& quadOutIndices = *d_quadOutIndices.beginWriteOnly();
+        SetIndex& tetrahedronOutIndices = *d_tetrahedronOutIndices.beginWriteOnly();
+        SetIndex& hexahedronOutIndices = *d_hexahedronOutIndices.beginWriteOnly();
+
+        // Write accessor for toplogical element in BOX
+        WriteOnlyAccessor< Data<VecCoord > > pointsInROI = d_pointsInROI;
+        WriteOnlyAccessor< Data<vector<Edge> > > edgesInROI = d_edgesInROI;
+        WriteOnlyAccessor< Data<vector<Triangle> > > trianglesInROI = d_trianglesInROI;
+        WriteOnlyAccessor< Data<vector<Quad> > > quadInROI = d_quadsInROI;
+        WriteOnlyAccessor< Data<vector<Tetra> > > tetrahedraInROI = d_tetrahedraInROI;
+        WriteOnlyAccessor< Data<vector<Hexa> > > hexahedraInROI = d_hexahedraInROI;
+        WriteOnlyAccessor< Data<VecCoord > > pointsOutROI = d_pointsOutROI;
+        WriteOnlyAccessor< Data<vector<Edge> > > edgesOutROI = d_edgesOutROI;
+        WriteOnlyAccessor< Data<vector<Triangle> > > trianglesOutROI = d_trianglesOutROI;
+        WriteOnlyAccessor< Data<vector<Quad> > > quadsOutROI = d_quadsOutROI;
+        WriteOnlyAccessor< Data<vector<Tetra> > > tetrahedraOutROI = d_tetrahedraOutROI;
+        WriteOnlyAccessor< Data<vector<Hexa> > > hexahedraOutROI = d_hexahedraOutROI;
+
+        // Clear lists
+        indices.clear();
+        edgeIndices.clear();
+        triangleIndices.clear();
+        quadIndices.clear();
+        tetrahedronIndices.clear();
+        hexahedronIndices.clear();
+        indicesOut.clear();
+        edgeOutIndices.clear();
+        triangleOutIndices.clear();
+        quadOutIndices.clear();
+        tetrahedronOutIndices.clear();
+        hexahedronOutIndices.clear();
+
+        pointsInROI.clear();
+        edgesInROI.clear();
+        trianglesInROI.clear();
+        quadInROI.clear();
+        tetrahedraInROI.clear();
+        hexahedraInROI.clear();
+        pointsOutROI.clear();
+        edgesOutROI.clear();
+        trianglesOutROI.clear();
+        quadsOutROI.clear();
+        tetrahedraOutROI.clear();
+        hexahedraOutROI.clear();
+
+
+        if (d_X0.getValue().size() == 0)
+        {
+            msg_warning() << "No rest position yet defined. Box might not work properly. \n"
+                            "This may be caused by an early call of init() on the box before  \n"
+                            "the mesh or the MechanicalObject of the node was initialized too";
```

Comment:
```suggestion
            msg_warning() << "No rest position yet defined. ROI might not work properly. \n"
                            "This may be caused by an early initialization of the ROI before  \n"
                            "the mesh or the MechanicalObject of the node was initialized too";
```